interface-impl-amf/src/main/java/org/mule/amf/impl/loader/ExchangeDependencyResourceLoader.java

```diff
@@ -12,26 +12,32 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import amf.client.execution.ExecutionEnvironment;
 import amf.client.remote.Content;
 import amf.client.resource.FileResourceLoader;
 import amf.client.resource.ResourceLoader;
 
 public class ExchangeDependencyResourceLoader implements ResourceLoader {
 
   private final File workingDir;
-  private final FileResourceLoader resourceLoader = new FileResourceLoader();
+  private final FileResourceLoader resourceLoader;
 
   private static final Pattern DEPENDENCY_PATH_PATTERN = Pattern.compile("^exchange_modules/|/exchange_modules/");
 
-  public ExchangeDependencyResourceLoader(String rootDir) {
+  public ExchangeDependencyResourceLoader(String rootDir, ExecutionEnvironment executionEnvironment) {
     String basePath = rootDir != null ? rootDir : ".";
     this.workingDir = new File(basePath);
+    this.resourceLoader = new FileResourceLoader(executionEnvironment);
   }
 
   @Override
   public CompletableFuture<Content> fetch(String path) {
-    if (path == null || path.isEmpty())
-      return fail();
+    final CompletableFuture<Content> future = new CompletableFuture<>();
+
+    if (path == null || path.isEmpty()) {
+      future.completeExceptionally(new Exception("Failed to apply."));
```

Comment:
why not throw Exception ?

Reply:
https://www.mulesoft.org/jira/browse/APIKIT-2599